### DISCUSSION TITLE
Coregistration of T1w and T2w for charm

### DISCUSSION THREAD

**Discussion by michefabb** [simnibs/simnibs#245] [7.Creating head models] [answered]
Dear SimNIBS experts,

I have the T1w and T2w_FLAIR of the same subject, I have coregistered them with FreeSurfer
```
mri_coreg
mri_vol2vol
```
and checking in Freeview they are now correctly coregistered.
Also  checking the charm_report.html, the "T2 registration overlay" looks fine:
![Screenshot 2023-11-23 alle 15 16 48](https://github.com/simnibs/simnibs/assets/128243878/5a7790a9-0ad3-4a70-9537-5879ab9b288e)

![Screenshot 2023-11-23 alle 15 16 58](https://github.com/simnibs/simnibs/assets/128243878/b9cf6740-3847-4c73-9301-fd29133178de)

However when running charm it looks they remain as before coregistration, in fact if I check the "coregistered template":

![Screenshot 2023-11-23 alle 15 05 48](https://github.com/simnibs/simnibs/assets/128243878/d9a10d96-d4bd-4946-ada0-b86c8db18a09)

leading to the following result:
![Screenshot 2023-11-23 alle 15 10 28](https://github.com/simnibs/simnibs/assets/128243878/163c7484-81d9-43ab-b8fe-2c4ba712240e)

I have tried already what suggested in the Tutorial [https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_affine_registration.html#fix-affine-registration-tutorial](url) and in this discussion #131 [https://github.com/simnibs/simnibs/discussions/131#discussioncomment-5636749](url) , both without success.

Would you kindly suggest how it could be fixed?

Thanks, regards,

M

**Comment by michefabb** [comment]
Dear all, would you kindly have any feedback? Appreciated!

**Reply by oulap** [reply, marked as answer]
Sorry for the late reply. 

Looks like the subjects you're working with are quite young. As a results the template does not really match the head anatomy (as it has been built from adult subjects).  To improve the results you could try to modify this line in the settings file:

https://github.com/simnibs/simnibs/blob/master/simnibs/charm.ini#L60

and set it to, for example:

affine_scales = [[0.8, 0.8, 0.8]]

to scale the template down. This is not guaranteed to fix the segmentation, because the template anatomy might be too far from that of your subjects, but it might improve it.

**Reply by michefabb** [reply]
Dear @oulap ,

thanks for your suggestion and apologize for my late reply.
I changed the settings.ini as you suggested ( `affine_scales = [[0.8, 0.8, 0.8]]` ) and this improved a lot the outcome; I attach it here (upper figure: charm outcome before applyng your settings; lower figure: charm outcome after using your settings).

![COMPARISON](https://github.com/simnibs/simnibs/assets/128243878/39c85940-cdac-4f85-87e3-44f1ed274864)

I noticed some issue on the scalp (left ear and some sort of "sinking" on the upper part, see picture) and skull (upper part), this is not mandatory, however if possible I would try to correct it: would you have any other suggestion to try to improve it even more?

![scalp](https://github.com/simnibs/simnibs/assets/128243878/f1e16e00-cdc2-4668-a6ca-45b776e7606d)

![skull](https://github.com/simnibs/simnibs/assets/128243878/e4f48516-25a3-4c85-8215-61a5441b8faa)



Greatly appreciated! Thanks, regards,

Mic

**Comment by oulap** [comment]
The ear is probably tricky to fix with just changing the affine. It seems there's a bit of mislabeling of the skull left (the bump in the last figure). You could try to go even lower on the affine scales to initialize the atlas even more shrunk. Otherwise you can always fix this manually using the directions here:

https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_headmodel.html

**Reply by michefabb** [reply]
Dear @oulap ,

apologize for late reply. I changed the settings.ini lowering even more the affine scales values up to 0.73 (for lower values charm returns an error of shape mismatch):
`affine_scales = [[0.73, 0.73, 0.73]] `
this fixed the issue, see pictures below respect to those of the previous post.

Appreciated your precious support!

Mic

![Screenshot 2024-01-11 alle 10 57 02](https://github.com/simnibs/simnibs/assets/128243878/fc61bda0-f12a-4533-9b7c-fc6a89ef90c9)
![Screenshot 2024-01-11 alle 10 56 18](https://github.com/simnibs/simnibs/assets/128243878/38fb3159-1cfc-4fa6-a652-145b41dd1797)
